file_catalog/server.py

```diff
@@ -292,28 +317,35 @@ def post(self):
 
 class SingleFileHandler(APIHandler):
     def initialize(self, **kwargs):
+        self.config = kwargs.pop('config')
         super(SingleFileHandler, self).initialize(**kwargs)
         self.files_url = os.path.join(self.base_url,'files')
 
     @catch_error
     @coroutine
     def get(self, mongo_id):
-        ret = yield self.db.get_file({'mongo_id':mongo_id})
-        if ret:
-            ret['_links'] = {
-                'self': {'href': os.path.join(self.files_url,mongo_id)},
-                'parent': {'href': self.files_url},
-            }
-
-            self.write(ret)
-        else:
-            self.send_error(404, message='not found')
+        try:
+            ret = yield self.db.get_file({'mongo_id':mongo_id})
+    
+            if ret:
+                ret['_links'] = {
+                    'self': {'href': os.path.join(self.files_url,mongo_id)},
+                    'parent': {'href': self.files_url},
+                }
+    
+                self.write(ret)
+            else:
+                self.send_error(404, message='not found')
+        except pymongo.errors.InvalidId:
+            self.send_error(400, message='`%s` is not a valid mongo_id, it must be a 12-byte input or a 24-character hex string' % mongo_id)
```

Comment:
Don't echo input back to the user. This is actually a security corner case that probably doesn't matter to us, but is good policy.  So, just say "not a valid mongo_id".

Same goes for all subsequent errors.